compiler/codegen/src/compile.rs

```diff
@@ -2698,11 +2695,191 @@ impl Compiler {
+                                        //    .collect();
+                                        let value = element
+                                            .format_spec
+                                            .as_ref()
+                                            .and_then(|x| {
+                                                self.source_code.text.get(
+                                                    x.range.start().to_usize()
+                                                        ..x.range.end().to_usize(),
+                                                )
+                                            })
+                                            .unwrap_or_default()
+                                            .to_owned();
+                                        self.emit_load_const(ConstantData::Str { value });
+
+                                        self.compile_expression(&element.expression)?;
+
+                                        emit!(
+                                            self,
+                                            Instruction::FormatValue {
+                                                conversion: element.conversion
+                                            }
+                                        );
+                                        args += 1;
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+
+                emit!(self, Instruction::BuildList { size: args });
+
+                let call = CallType::Positional { nargs: 1 };
+                self.compile_method_call(call)
+            }
+            Expr::StringLiteral(string) => {
+                self.emit_load_const(ConstantData::Str {
+                    value: string.value.to_str().to_owned(),
+                });
+            }
+            Expr::BytesLiteral(bytes) => {
+                let iter = bytes.value.iter().flat_map(|x| x.iter().copied());
+                let v: Vec<u8> = iter.collect();
+                self.emit_load_const(ConstantData::Bytes { value: v });
+            }
+            Expr::NumberLiteral(number) => match &number.value {
+                Number::Int(int) => {
+                    let value = if let Some(small) = int.as_u64() {
+                        BigInt::from(small)
+                    } else {
+                        parse_big_integer(int).map_err(|e| self.error(e))?
+                    };
+                    self.emit_load_const(ConstantData::Integer { value });
+                }
+                Number::Float(float) => {
+                    self.emit_load_const(ConstantData::Float { value: *float });
+                }
+                Number::Complex { real, imag } => {
+                    self.emit_load_const(ConstantData::Complex {
+                        value: Complex::from_polar(*real, *imag),
```

Comment:
```suggestion
                        value: Complex::new(*real, *imag),
```
we have plane coord. this will fix `cargo run extra_tests/snippets/builtin_complex.py`

Reply:
Thanks, fixed in #5494